parlai/core/metrics.py

```diff
@@ -259,20 +259,23 @@ class MacroAverageMetric(Metric):
     AverageMetrics already.
     """
 
-    __slots__ = ('_values',)
+    __slots__ = ('_values', '_names')
 
-    def __init__(self, metrics: List[Metric]) -> None:
+    def __init__(self, metrics: List[Metric], names: List[str]) -> None:
```

Comment:
Is there any advantage to having two Lists as args instead of one arg that's Dict[str, Metric]? Could be a cleaner way to associate metrics with names

Reply:
You're right. I was feeling worried about it yesterday for some reason I don't know, but this is much better.